### ISSUE TITLE
ctwill and table of contents ToC

### ISSUE THREAD

**Issue by joaocortes** [ascherer/cwebbin#45]
`ctwill` does not produce a ToC, even with the option `+x`. According to the `man` documentation,

```
       • -x: omit indices, section names, table of contents
```
Running `ctwill-twinx attempt.tex > index.tex` creates a master index, but there is no table of contents.

Is this by design?

**Comment by ascherer** [comment]
Yes, it is by design. See http://ftp.cs.stanford.edu/pub/ctwill/ctwimac.tex and https://github.com/ascherer/cwebbin/blob/master/texinputs/pdfctwimac.tex for details about `\inx`, `\fin`, and `con`. If you use `ctwill +pdf`, at least you should get "bookmarks" as a table-of-contents.

**Comment by ascherer** [comment]
In fact, the `ctwill` executable – just like its sibling `cweave` – _does_ create files `jobname.idx`, `jobname.scn`, and `jobname.toc` (replace `jobname` appropriately). It's just tthe TeX macros that are not elaborate enough to process these informations. (Personally, I never came around to even learn the usage of `ctwill-twinx` to generate a “master index” for `ctwill.w` and `common.w`.)

**Comment by joaocortes** [comment]
Thanks. I got it after looking at the different definitions of those macros. When using `ctwill`, `\con` is never defined. It seems to be a strange choice. I can't think of any reason why `ctwill` should have an harder time building the ToC. I will try to write an extension, as soon as I have time to do that.

**Comment by ascherer** [comment]
You may want to look at [CWEB 4.3 dev](https://github.com/ascherer/cwebbin/tree/cweb-4.3).  The extended `pdfctwimac.tex` already produces a table-of-contents
- at the end of the PDF
- it does not yet cooperate with `pdfwebtocfront.tex`
- all page numbers are **0**

**Comment by joaocortes** [comment]
Hi again. I am trying to get this solved. I looked into the current version of`pdfctwimac.tex`, and it seems it still does not produce a ToC. Did you reference the wrong branch on your last comment?

**Comment by ascherer** [comment]
The current state of affairs is [CWEB 4.8 dev](https://github.com/ascherer/cweb/tree/cweb-4.8-dev) and [CWEBbin 4.8 dev](https://github.com/ascherer/cwebbin/tree/cweb-4.8). Table-of-Contents in CTWILL was not on the top of the list.
